presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionSymbolInliner.java

```diff
@@ -51,8 +62,19 @@ public Expression rewriteSymbolReference(SymbolReference node, Void context, Exp
         @Override
         public Expression rewriteLambdaExpression(LambdaExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
-            // Lambda does not support capture yet. As a result, relation/columns can not exist in lambda.
-            return node;
+            List<String> addedNames = new ArrayList<>();
+            for (LambdaArgumentDeclaration argument : node.getArguments()) {
```

Comment:
maintaining `excludedNames` set is tricky and not very recursive friendly.

As I understand this is because there might be two `SymbolReferences` with the same name, one of them is lambda argument reference and the other is normal column/channel reference.
When this would be the case? `SymbolAllocator` should always return unique symbol name, therefore there shouldn't be clash between those two worlds.